src/storage/src/hummock/store/version.rs

```diff
@@ -275,15 +273,7 @@ impl HummockReadVersion {
                 // TODO: add a check to ensure that the added batch id of added imm is greater than
                 // the batch id of imm at the front
                 StagingData::ImmMem(imm) => {
-                    if let Some(item) = self.staging.imm.front() {
-                        // check batch_id order from newest to old
-                        debug_assert!(item.batch_id() < imm.batch_id());
```

Comment:
I think the assumption that newer imm will have larger batch id still holds. Why do we remove this assertion?

Reply:
I have add it back